k8s.gcr.io/images/k8s-staging-autoscaling/add-version.sh

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# Adds a new component version for promotion.
+# Can also be used to verify if an already-added version is OK, should produce
+# empty diff in such a case.
+# NOTE: Only modifies local files, does not create a PR.
+
```

Comment:
`set +o errexit`

to die if there's an error retuned by one of commands?

Reply:
I also suggest `set -o pipefail`